modules/authentication/services/index.js

```diff
@@ -8,7 +8,7 @@ function checkAuth(store) {
     return store.getHydratedState()
       .then((state) => {
         const hasAllTokenInfoKeys = VALID_TOKEN_INFO_FIELDS.filter((field) => {
-          return state.authentication.tokenInfo[field];
+          return state.getIn(['authentication', 'tokenInfo', field]);
```

Comment:
so we're using immutable for the root level state? I'm sure you already discussed the pros and cons of doing so, are we roughly aware of what those are?

Reply:
My understanding is that it's just more inline with the documentation around using Immutable.JS with Redux. It's [this](https://redux.js.org/recipes/using-immutable.js-with-redux#make-your-entire-redux-state-tree-an-immutablejs-object) page in particular but for some reason it's not loading the CSS for me so I'll just paste a screenshot

<img width="827" alt="screen shot 2018-02-19 at 3 56 50 pm" src="https://user-images.githubusercontent.com/10897808/36396719-8a2cd258-158d-11e8-9345-052dfa8d5174.png">

I guess it's just that if you're already using Immutable.JS for all your reducers just might as well make your top level Immutable too